tests/unit/s2n_x509_validator_custom_oids_test.c

```diff
@@ -0,0 +1,208 @@
+/*
+* Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+*
+* Licensed under the Apache License, Version 2.0 (the "License").
+* You may not use this file except in compliance with the License.
+* A copy of the License is located at
+*
+*  http://aws.amazon.com/apache2.0
+*
+* or in the "license" file accompanying this file. This file is distributed
+* on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+* express or implied. See the License for the specific language governing
+* permissions and limitations under the License.
+*/
+
+#include "crypto/s2n_libcrypto.h"
+#include "s2n_test.h"
+#include "testlib/s2n_testlib.h"
+
+#define S2N_SINGLE_OID_CERT_CHAIN "../pems/custom_oids/single_oid_cert_chain.pem"
+#define S2N_SINGLE_OID_KEY        "../pems/custom_oids/single_oid_key.pem"
+
+#define S2N_MULTIPLE_OIDS_CERT_CHAIN "../pems/custom_oids/multiple_oids_cert_chain.pem"
+#define S2N_MULTIPLE_OID_KEY         "../pems/custom_oids/multiple_oids_key.pem"
+
+const char *single_oid[] = { "1.3.178.25240.2" };
+const char *multiple_oids[] = { "1.3.178.25240.2", "1.3.178.25240.3" };
+const int multiple_oid_count = s2n_array_len(multiple_oids);
+
+int main(int argc, char *argv[])
+{
+    BEGIN_TEST();
+
+    {
+        /* Safety Check */
+        EXPECT_FAILURE_WITH_ERRNO(s2n_config_set_custom_critical_oids(NULL, single_oid, 1), S2N_ERR_NULL);
+
+        struct s2n_config *test_config = s2n_config_new_minimal();
+        EXPECT_NOT_NULL(test_config);
+
+#ifndef S2N_LIBCRYPTO_SUPPORTS_CUSTOM_OID
+        EXPECT_FAILURE_WITH_ERRNO(s2n_config_set_custom_critical_oids(test_config, single_oid, 1),
+                S2N_ERR_INTERNAL_LIBCRYPTO_ERROR);
+        EXPECT_SUCCESS(s2n_config_free(test_config));
+
+        END_TEST();
+#endif
+
+        EXPECT_SUCCESS(s2n_config_set_custom_critical_oids(test_config, single_oid, 1));
+        /* invoke again to reset custom oids */
+        EXPECT_SUCCESS(s2n_config_set_custom_critical_oids(test_config, multiple_oids, multiple_oid_count));
+        EXPECT_EQUAL(sk_ASN1_OBJECT_num(test_config->custom_crit_oids), multiple_oid_count);
+        EXPECT_SUCCESS(s2n_config_free(test_config));
```

Comment:
It's typically easier to reason about tests and update them later when they're individually scoped, and variables aren't shared/modified between them (like test_config is here). I'd recommend creating a new scope for each test and just recreating the config in each of them.